contracts/account/extensions/AccountERC7579.sol

```diff
@@ -259,14 +268,16 @@ abstract contract AccountERC7579 is Account, IERC1271, IERC7579Execution, IERC75
     /**
      * @dev Uninstalls a module of the given type with the given de-initialization data.
      *
-     * For the fallback module type, the `deInitData` is expected to be a tuple of a 4-byte selector and the
-     * rest of the data to be sent to the handler when calling {IERC7579Module-onUninstall}.
+     * For the fallback module type, the `deInitData` is expected to be the (packed) concatenation of a 4-byte
+     * selector and the rest of the data to be sent to the handler when calling {IERC7579Module-onUninstall}.
      *
      * Requirements:
      *
      * * Module must be already installed. Reverts with {ERC7579UninstalledModule} otherwise.
      */
     function _uninstallModule(uint256 moduleTypeId, address module, bytes memory deInitData) internal virtual {
+        require(supportsModule(moduleTypeId), ERC7579Utils.ERC7579UnsupportedModuleType(moduleTypeId));
```

Comment:
Thoughts on reverting in the else clause instead of checking here?

Reply:
I was thinking "what if someone wants to disable a type of module". For example, keep validation module but stop supporting execution module.

- They would update the `supportsModule` function.
- That would prevent installing
- If they try to uninstall, they would get a `ERC7579UninstalledModule` error instead of a `ERC7579UnsupportedModuleType`.

I don't have strong opinion either way. @ernestognw wdyt ?

Reply:
Note that `supportsModule` doesn't perform any storage operation.

Reply:
That's reasonable.

Reply:
I slightly prefer to call `supportsModule` first as it's implemented now.